opal/Makefile.am

```diff
@@ -53,8 +53,17 @@ lib@OPAL_LIB_PREFIX@open_pal_la_LIBADD = \
         util/libopalutil.la \
 	$(MCA_opal_FRAMEWORK_LIBS)
 lib@OPAL_LIB_PREFIX@open_pal_la_DEPENDENCIES = $(lib@OPAL_LIB_PREFIX@open_pal_la_LIBADD)
+
+if HAVE_ATTRIBUTE_DESTRUCTOR
+
 lib@OPAL_LIB_PREFIX@open_pal_la_LDFLAGS = -version-info $(libopen_pal_so_version)
 
+else
+
+lib@OPAL_LIB_PREFIX@open_pal_la_LDFLAGS = -version-info $(libopen_pal_so_version) -Wl,-fini,opal_cleanup
```

Comment:
Ah, yes. Good point. There may be compiler/linker combinations that will not support this. I have no problem leaking memory on those systems. It is only the constructor/destructor tables.